api/src/main/java/org/apache/iceberg/Scan.java

```diff
@@ -77,6 +78,21 @@ public interface Scan<ThisT, T extends ScanTask, G extends ScanTaskGroup<T>> {
    */
   ThisT includeColumnStats();
 
+  /**
+   * Create a new scan from this that loads the column stats for the specific columns with each data
+   * file. If the columns set is empty or <code>null</code> then all column stats will be kept, if
+   * {@link #includeColumnStats()} is set.
+   *
+   * <p>Column stats include: value count, null value count, lower bounds, and upper bounds.
+   *
+   * @param columnsToKeepStats column ids from the table's schema
```

Comment:
This API does not use column IDs, it uses column names. This method should not accept a collection of IDs and should accept strings and convert instead.

Also, the convention is to use `Collection<String>` rather than `Set` and to add a `String...` method for convenience.

Reply:
+1.

We have `select` that accepts `Collection<String> columns` below. I understand using IDs is straightforward but someone has to do the conversion from name to ID and we better do it ourselves behind this API. If an external long-running process needs to ensure the ID did not change, it can it do it outside.

Reply:
+1 on using string